File: FlashCards.Models/Card.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FlashCards.Models
{
    public class Card : BaseEntity
    {
        [ForeignKey("DeckId")]
        public Guid DeckId { get; set; }

        public required string Question { get; set; }

        public required string Answer { get; set; }

        public DateTime LastReviewed { get; set; }

        public RatingStatus LastStatus { get; set; }

        public int ReviewCount { get; set; } = 0;

        public double EaseFactor { get; set; } = 2.5;

        public int Interval { get; set; } = 1;

        public DateTime NextReviewDate { get; set; }

        public ICollection<Review> Reviews { get; set; } = [];
    }
}

File: FlashCards.UI/src/app/shared/models/Card.ts
import { UUIDTypes } from 'uuid';

export type Card = {
  id?: UUIDTypes;
  deckId: UUIDTypes;
  question: string;
  answer: string;
  lastReviewed: Date;
  lastStatus: RatingStatus;
  reviewCount: number;
};

export enum RatingStatus {
  Forgot = 1,
  Hard = 2,
  Good = 3,
  Easy = 4,
}
